packages/lib/server-only/attachment/set-document-attachments.ts

```diff
@@ -0,0 +1,110 @@
+import type { Attachment, User } from '@prisma/client';
+
+import type { RequestMetadata } from '@documenso/lib/universal/extract-request-metadata';
+import { prisma } from '@documenso/prisma';
+
+import { AppError } from '../../errors/app-error';
+import { AppErrorCode } from '../../errors/app-error';
+import { DOCUMENT_AUDIT_LOG_TYPE } from '../../types/document-audit-logs';
+import { createDocumentAuditLogData } from '../../utils/document-audit-logs';
+
+export type CreateAttachmentsOptions = {
+  documentId: number;
+  attachments: Pick<Attachment, 'id' | 'label' | 'url' | 'type'>[];
+  user: Pick<User, 'id' | 'email' | 'name'>;
+  requestMetadata: RequestMetadata;
+};
+
+export const setDocumentAttachments = async ({
+  documentId,
+  attachments,
+  user,
+  requestMetadata,
+}: CreateAttachmentsOptions) => {
+  const document = await prisma.document.findUnique({
+    where: {
+      id: documentId,
```

Comment:
Need the team where query here otherwise you can update the attachment for any document with a crafted request.